include/node.tcc

```diff
@@ -346,6 +347,63 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_velocity_constraints() {
   }
 }
 
+// !Apply absorbing constraints
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::apply_absorbing_constraint(
+    unsigned dir, double delta, double h_min, double a, double b) {
+  bool status = true;
+  try {
+
+    if (dir >= Tdim) {
+      throw std::runtime_error("Direction is out of bounds");
+      status = false;
+    }
+    if (material_ids_.size() != 0) {
+      // Get material id
+      auto mat_id = material_ids_.begin();
+
+      // Extract material properties and displacements
+      double pwave_v = this->property_handle_->property(
+          "wave_velocities", prop_id_, *mat_id, Tdim)(0);
+      double swave_v = this->property_handle_->property(
+          "wave_velocities", prop_id_, *mat_id, Tdim)(1);
+      double density =
+          this->property_handle_->property("density", prop_id_, *mat_id)(0);
+      Eigen::Matrix<double, Tdim, 1> material_displacement =
+          this->property_handle_->property("displacements", prop_id_, *mat_id,
+                                           Tdim);
+
+      // Wave velocity Eigen Matrix
+      Eigen::Matrix<double, Tdim, 1> wave_velocity =
+          Eigen::MatrixXd::Constant(Tdim, 1, b * swave_v);
+      wave_velocity(dir, 0) = a * pwave_v;
+      // Spring constant Eigen matrix
+      double k_s = (density * pow(swave_v, 2)) / delta;
+      double k_p = (density * pow(pwave_v, 2)) / delta;
+      Eigen::Matrix<double, Tdim, 1> spring_constant =
+          Eigen::MatrixXd::Constant(Tdim, 1, k_s);
+      spring_constant(dir, 0) = k_p;
+
+      // Iterate through each phase
+      for (unsigned phase = 0; phase < Tnphases; ++phase) {
+        // Calculate Aborbing Traction
+        Eigen::Matrix<double, Tdim, 1> absorbing_traction_ =
+            this->velocity_.col(phase).cwiseProduct(wave_velocity) * density +
+            material_displacement.cwiseProduct(spring_constant);
+
+        // Update external force
+        this->update_external_force(true, phase, -absorbing_traction_);
```

Comment:
@cgeudeker @kks32 
`absorbing_traction_` is going to be [ force / (length^2) ] but we should be passing just [ force ] to `update_external_force(..)`. 

Using `h_min` and unit thickness into the page I think we can easily go from traction -> force. But we need to think about how to handle the corner nodes where the nodal influence is only `h_min / 2`